src/plugins/data/public/ui/filter_bar/filter_item.tsx

```diff
@@ -349,32 +350,44 @@ export function FilterItem(props: FilterItemProps) {
     return null;
   }
 
-  const badge = (
-    <FilterView
-      filter={filter}
-      valueLabel={valueLabelConfig.title}
-      filterLabelStatus={valueLabelConfig.status}
-      errorMessage={valueLabelConfig.message}
-      className={getClasses(filter.meta.negate ?? false, valueLabelConfig)}
-      iconOnClick={() => props.onRemove()}
-      onClick={handleBadgeClick}
-      data-test-subj={getDataTestSubj(valueLabelConfig)}
-    />
-  );
+  const filterViewProps = {
+    filter,
+    valueLabel: valueLabelConfig.title,
+    filterLabelStatus: valueLabelConfig.status,
+    errorMessage: valueLabelConfig.message,
+    className: getClasses(filter.meta.negate ?? false, valueLabelConfig),
+    iconOnClick: () => props.onRemove(),
+    onClick: handleBadgeClick,
+    ['data-test-subj']: getDataTestSubj(valueLabelConfig),
+    readonly: props.readonly,
+  };
+
+  const popoverProps: CommonProps & HTMLAttributes<HTMLDivElement> & EuiPopoverProps = {
+    id: `popoverFor_filter${id}`,
+    className: `globalFilterItem__popover`,
+    anchorClassName: `globalFilterItem__popoverAnchor`,
+    isOpen: isPopoverOpen,
+    closePopover: () => {
+      setIsPopoverOpen(false);
+    },
+    button: <FilterView {...filterViewProps} />,
+    panelPaddingSize: 'none',
+  };
+
+  if (props.readonly) {
+    return (
+      <EuiPopover
+        panelClassName="globalFilterItem__readonlyPanel"
+        anchorPosition="upCenter"
+        {...popoverProps}
+      >
+        <FilterView {...filterViewProps} hideAlias />
```

Comment:
Hmm, I'm not a huge fan of this change. Instead of having a button that just opens a popover with the same button (except it shows the text instead of the alias) doesn't seem very useful. What if, in readonly state, we actually open the filter editor but in a disabled state? (Then we can get rid of the `hideAlias` flag.)

Reply:
sounds like a good change

Reply:
Discussed offline, came up with creating follow up issue to implement disabled state of filter editor.

Reply:
Here is the issue #127870